core/src/main/scala/shapeless/ops/hlists.scala

```diff
@@ -3030,4 +3030,57 @@ object hlist {
         def apply(l: HNil): Out = HNil
       }
   }
+
+  /**
+   * Type class supporting mappings from type `T` to an `HList`. Currently only supports mapping nested pairs to an `HList`
+   *
+   * @author Michael Zuber
+   */
+  sealed trait ToHList[-T] {
```

Comment:
```suggestion
  sealed trait ProductToHList[T] extends DepFn1[T] {
```
I think we should be consistent with #699 

@mgzuber I am also happy to make those changes if it's too late for you.